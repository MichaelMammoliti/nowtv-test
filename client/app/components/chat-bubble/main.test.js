import React from 'react';
import { shallow } from 'enzyme';
import ChatBubble from './main.js';

const getProps = () => {
  return {
    id: 'b03569ae-ccbf-4975-8040-4daba638b407',
    userId: '16373df5-da0a-4074-8295-f916b94269f4',
    message: 'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
    timestamp: '2016-11-09T05:04:58Z',
    sender: {
      id: 'e837c9f5-247f-445f-bcc3-7d434348336b',
      firstName: 'Martin',
      lastName: 'Bradley',
      email: 'mbradley0@google.it',
      avatar: 'http://dummyimage.com/100x100.png/5fa2dd/ffffff',
      ip: '166.124.172.160',
    },
  };
};

const getUnit = (props = getProps()) =>
  shallow(<ChatBubble {...props} />)
;

describe('ChatBubble', () => {
  it('should render correctly', () => {
    const unit = getUnit();

    expect(unit).toMatchSnapshot();
  });

  it('should render an image when the "sender.avatar" prop is set', () => {
    const props = getProps();
    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__avatar').exists()).toBe(true);
  });

  it('should NOT render an image when the "sender.avatar" prop is NOT set', () => {
    const props = getProps();
    delete props.sender.avatar;

    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__avatar').exists()).toBe(false);
  });

  it('should render a message when the "message" prop is set', () => {
    const props = getProps();
    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__message').exists()).toBe(true);
  });

  it('should NOT render a message when the "message" prop is NOT set', () => {
    const props = getProps();
    props.message = undefined;

    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__message').exists()).toBe(false);
  });

  it('should render a date when the "timestamp" prop is set in a formatted way', () => {
    const props = getProps();
    props.timestamp = '2016-11-09T05:04:58Z';
    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__date').text()).toBe('09 Nov 2016 - 05:04');
  });

  it('should NOT render a date when the "timestamp" prop is NOT set', () => {
    const props = getProps();
    delete props.timestamp;

    const unit = getUnit(props);

    expect(unit.find('.chat-bubble__date').exists()).toBe(false);
  });
});