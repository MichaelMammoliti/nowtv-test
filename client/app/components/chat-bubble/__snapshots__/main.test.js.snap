// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatBubble should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatBubble
    id="b03569ae-ccbf-4975-8040-4daba638b407"
    message="Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus."
    sender={
      Object {
        "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        "email": "mbradley0@google.it",
        "firstName": "Martin",
        "id": "e837c9f5-247f-445f-bcc3-7d434348336b",
        "ip": "166.124.172.160",
        "lastName": "Bradley",
      }
    }
    timestamp="2016-11-09T05:04:58Z"
    userId="16373df5-da0a-4074-8295-f916b94269f4"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="chat-bubble__body"
        >
          <div
            className="chat-bubble__avatar"
          >
            <img
              alt="profile"
              src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
            />
          </div>
          <div
            className="chat-bubble__message"
          >
            <span>
              Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
            </span>
          </div>
        </div>,
        <div
          className="chat-bubble__date"
        >
          <span>
            09 Nov 2016 - 05:04
          </span>
        </div>,
        <div
          className="chat-bubble__email"
        >
          <span>
            mbradley0@google.it
          </span>
        </div>,
      ],
      "className": "chat-bubble",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="chat-bubble__avatar"
            >
              <img
                alt="profile"
                src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
              />
            </div>,
            <div
              className="chat-bubble__message"
            >
              <span>
                Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
              </span>
            </div>,
          ],
          "className": "chat-bubble__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="profile"
                src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
              />,
              "className": "chat-bubble__avatar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "profile",
                "src": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
              </span>,
              "className": "chat-bubble__message",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
              },
              "ref": null,
              "rendered": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            09 Nov 2016 - 05:04
          </span>,
          "className": "chat-bubble__date",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "09 Nov 2016 - 05:04",
          },
          "ref": null,
          "rendered": "09 Nov 2016 - 05:04",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            mbradley0@google.it
          </span>,
          "className": "chat-bubble__email",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mbradley0@google.it",
          },
          "ref": null,
          "rendered": "mbradley0@google.it",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="chat-bubble__body"
          >
            <div
              className="chat-bubble__avatar"
            >
              <img
                alt="profile"
                src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
              />
            </div>
            <div
              className="chat-bubble__message"
            >
              <span>
                Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
              </span>
            </div>
          </div>,
          <div
            className="chat-bubble__date"
          >
            <span>
              09 Nov 2016 - 05:04
            </span>
          </div>,
          <div
            className="chat-bubble__email"
          >
            <span>
              mbradley0@google.it
            </span>
          </div>,
        ],
        "className": "chat-bubble",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="chat-bubble__avatar"
              >
                <img
                  alt="profile"
                  src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
                />
              </div>,
              <div
                className="chat-bubble__message"
              >
                <span>
                  Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
                </span>
              </div>,
            ],
            "className": "chat-bubble__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="profile"
                  src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
                />,
                "className": "chat-bubble__avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "profile",
                  "src": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
                </span>,
                "className": "chat-bubble__message",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                },
                "ref": null,
                "rendered": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              09 Nov 2016 - 05:04
            </span>,
            "className": "chat-bubble__date",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "09 Nov 2016 - 05:04",
            },
            "ref": null,
            "rendered": "09 Nov 2016 - 05:04",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              mbradley0@google.it
            </span>,
            "className": "chat-bubble__email",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mbradley0@google.it",
            },
            "ref": null,
            "rendered": "mbradley0@google.it",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
